src/InvoiceNumber/Strategy/Infinite.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Drupal\commerce_invoice\InvoiceNumber\Strategy;
+
+class Infinite extends DatabaseStrategyBase {
```

Comment:
This is not actually infinite (limited by 32 bits), so there's probably a better name like `Consecutive`